x/ibc/02-client/handler.go

```diff
@@ -10,43 +10,24 @@ import (
 	"github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
 	"github.com/cosmos/cosmos-sdk/x/ibc/02-client/keeper"
 	"github.com/cosmos/cosmos-sdk/x/ibc/02-client/types"
-	localhosttypes "github.com/cosmos/cosmos-sdk/x/ibc/09-localhost/types"
 )
 
 // HandleMsgCreateClient defines the sdk.Handler for MsgCreateClient
-func HandleMsgCreateClient(ctx sdk.Context, k keeper.Keeper, msg exported.MsgCreateClient) (*sdk.Result, error) {
-	var (
-		consensusHeight uint64
-		clientState     exported.ClientState
-	)
-
-	switch msg.(type) {
-	// localhost is a special case that must initialize client state
-	// from context and not from msg
-	case *localhosttypes.MsgCreateClient:
-		clientState = localhosttypes.NewClientState(ctx.ChainID(), ctx.BlockHeight())
-		// Localhost consensus height is chain's blockheight
-		consensusHeight = uint64(ctx.BlockHeight())
-	default:
-		clientState = msg.InitializeClientState()
-		if consState := msg.GetConsensusState(); consState != nil {
-			consensusHeight = consState.GetHeight()
-		}
-	}
+func HandleMsgCreateClient(ctx sdk.Context, k keeper.Keeper, msg *types.MsgCreateClient) (*sdk.Result, error) {
+	clientState := types.MustUnpackClientState(msg.ClientState)
+	consensusState := types.MustUnpackConsensusState(msg.ConsensusState)
```

Comment:
These will panic if they fail, should we return errors instead?

Reply:
Even though these are checked in `ValidateBasic`, I think it is safer to check here in case those implementations change